pandas/tests/indexing/test_loc.py

```diff
@@ -1130,3 +1130,17 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
     tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
     tm.assert_frame_equal(df, df.loc[list(idx)])
+
+
+def test_loc_replace_subset_with_subset():
+    # GH#36424 Should raise a SettingWithCopyError
+    df1 = pd.DataFrame(
+        data=np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+        columns=["A", "B", "C"],
+        index=[0, 0, 1],
+    )
+    df2 = df1.copy()
+    df2[:] = np.nan
+    # It fail if a SettingWithCopyError is not raised
```

Comment:
Can remove this comment since it's understood

Reply:
Done